api/api-version-infos-v1/src/main/java/com/thoughtworks/go/apiv1/versioninfos/VersionInfosControllerV1.java

```diff
@@ -0,0 +1,116 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.versioninfos;
+
+import com.thoughtworks.go.api.ApiController;
+import com.thoughtworks.go.api.ApiVersion;
+import com.thoughtworks.go.api.representers.JsonReader;
+import com.thoughtworks.go.api.spring.ApiAuthenticationHelper;
+import com.thoughtworks.go.api.util.GsonTransformer;
+import com.thoughtworks.go.apiv1.versioninfos.models.GoLatestVersion;
+import com.thoughtworks.go.apiv1.versioninfos.representers.GoLatestVersionRepresenter;
+import com.thoughtworks.go.apiv1.versioninfos.representers.VersionInfoRepresenter;
+import com.thoughtworks.go.domain.VersionInfo;
+import com.thoughtworks.go.server.service.VersionInfoService;
+import com.thoughtworks.go.server.service.result.HttpLocalizedOperationResult;
+import com.thoughtworks.go.spark.Routes;
+import com.thoughtworks.go.spark.spring.SparkSpringController;
+import com.thoughtworks.go.util.SystemEnvironment;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import spark.Request;
+import spark.Response;
+
+import java.io.IOException;
+
+import static spark.Spark.*;
+
+@Component
+public class VersionInfosControllerV1 extends ApiController implements SparkSpringController {
+    private final ApiAuthenticationHelper apiAuthenticationHelper;
+    private VersionInfoService versionInfoService;
+    private SystemEnvironment systemEnvironment;
+
+    @Autowired
+    public VersionInfosControllerV1(ApiAuthenticationHelper apiAuthenticationHelper, VersionInfoService versionInfoService, SystemEnvironment systemEnvironment) {
+        super(ApiVersion.v1);
+        this.apiAuthenticationHelper = apiAuthenticationHelper;
+        this.versionInfoService = versionInfoService;
+        this.systemEnvironment = systemEnvironment;
+    }
+
+    @Override
+    public String controllerBasePath() {
+        return Routes.VersionInfos.BASE;
+    }
+
+    @Override
+    public void setupRoutes() {
+        path(controllerBasePath(), () -> {
+            before("", mimeType, this::setContentType);
+            before("/*", mimeType, this::setContentType);
+
+            before("", mimeType, this.apiAuthenticationHelper::checkUserAnd403);
+            before("/*", mimeType, this.apiAuthenticationHelper::checkUserAnd403);
+
+            get(Routes.VersionInfos.STALE, this::stale);
+            get(Routes.VersionInfos.LATEST_VERSION, this::latestVersion);
+            patch(Routes.VersionInfos.GO_SERVER, this::updateGoServerInfo);
```

Comment:
need to specify `mimeType`. Not specifying `mimeType` will cause the API to work for all the API versions.

Example: 
change:  `get(Routes.VersionInfos.STALE, this::stale);` 
to            `get(Routes.VersionInfos.STALE, mimeType, this::stale);`

Reply:
Done: https://github.com/gocd/gocd/pull/8359